packages/sdk/src/utils.ts

```diff
@@ -37,6 +37,16 @@ export function uuid(): string {
   return 'xxxxxxxx-xxxx-4xxx-xxxx-xxxxxxxxxxxx'.replace(/x/g, () => HEX_ALPHA[(Math.random() * 16) | 0]);
 }
 
+export function utf8ToBase64(str: string): string {
+  const encoder = new TextEncoder();
+  const bytes = encoder.encode(str);
+  let binary = '';
+  for (let i = 0; i < bytes.length; i++) {
+    binary += String.fromCharCode(bytes[i]);
+  }
```

Comment:
This would be a bit more idiomatic and efficient:
```suggestion
  const binary = String.fromCharCode(...bytes);
```